### PR TITLE
documentation about future plans

### PR DESCRIPTION
**Pull request by mm-near**

### CODE REVIEW

File: docs/architecture/next/malicious_chunk_producer_and_phase2.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+sends it to ``B1``, which includes it in the block.
+
+And here's where the complexity starts - as most of the valiators will NOT 
+track the shard which ``C1`` was producing - so they will still sign the block.
+
+The validators that do track that shard will of course (assuming that they are non-malicious) refuse the sign. But overall, they will be a small majority - so the block is going to get enough signatures and be added to the chain.
+
+### Challenges, Slashing and Rollbacks
+
+So we're in a pickle - as a malicious chunk was just added to the chain. And Fthat's why need to have mechanisms to automatically recover from such situations: Challenges, Slashing and Rollbacks.
```
Comment (nikurt) [inline docs/architecture/next/malicious_chunk_producer_and_phase2.md:44]:
```suggestion
So we're in a pickle - as a malicious chunk was just added to the chain. And
that's why need to have mechanisms to automatically recover from such situations:
Challenges, Slashing and Rollbacks.
```

File: docs/architecture/next/malicious_chunk_producer_and_phase2.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+When anyone notices that a current chain contains a wrong transition - they 
+submit such challenge to the next block producer, which can easily verify it 
+and it to the next block.
+
+Then the validators do the verification themselves, and if successful, they 
+sign the block.
+
+When such block is succesfully signed, the protocol automatically slashes 
+correct nodes (more details below) and initiates the rollback to bring the
```
Comment (nikurt) [inline docs/architecture/next/malicious_chunk_producer_and_phase2.md:63]:
```suggestion
malicious nodes (more details below) and initiates the rollback to bring the 
```

File: docs/architecture/next/malicious_chunk_producer_and_phase2.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+database (or a nasty bug in the code) - and the effect would be the same - the 
+chunk that you produced would be marked as malicious, and you'd lose your stake 
+(which will be a super-scary even for any legitimate validator).
+
+
+So the open question is - can we do something 'smarter' in the protocol to
+detect the case, where there is 'just a single' malicious (or buggy) chunk 
+producer and avoid the expensive rollback?
+
+(one strawman proposal - if we see that the previous block didn't contain any signatures from validators that were tracking a given shard - mark the block as failed and ignore the chunks from this shard for the next X blocks).
```
Comment (nikurt) [inline docs/architecture/next/malicious_chunk_producer_and_phase2.md:108]: Irrelevant in the context of this chapter.

**Review by nikurt — APPROVED** [review]

File: docs/architecture/next/malicious_chunk_producer_and_phase2.md (hunk 3 above)
Comment (mm-near) [reply docs/architecture/next/malicious_chunk_producer_and_phase2.md]: done.

**Pull request merged**